Makefile

```diff
@@ -0,0 +1,38 @@
+SHELL := /bin/bash
+DEST := result
+PORT := 5000
+
+.PHONY: all clean html web compile serve
+
+UNAME = $(shell uname)
+ifeq ($(UNAME), Linux)
+CP_ARGS = -l
+endif
+ifeq ($(UNAME), Darwin)
+CP_ARGS =
+endif
+
+all: serve
+
+prep:
+	mkdir -p $(DEST)/nightly
+
+clean:
+	$(MAKE) -C guides/ clean
+	rm -rf $(DEST) web/output/
+
+html: prep
+	$(MAKE) -C guides/ html BUILD=foreman-el
+	$(MAKE) -C guides/ html BUILD=foreman-deb
+	$(MAKE) -C guides/ html BUILD=katello
+
+web: prep
+	pushd web; bundle exec nanoc; popd
+
+compile: web html
+	cp $(CP_ARGS) -nr web/output/* $(DEST)
+	cp $(CP_ARGS) -nr guides/build/* $(DEST)/nightly/
+	for V in 3.2 3.1 3.0 2.5 2.4; do ln -sf nightly $(DEST)/$$V; done
```

Comment:
Yeah I know that it can introduce weird things, but you usually only need the stable branches to test the menu (the version switcher). You simply always build the current branch.

Updated and extracted into variable.